src/mmw/js/src/core/views.js

```diff
@@ -461,12 +483,53 @@ var MapView = Marionette.ItemView.extend({
 
     fitToAoi: function() {
         var areaOfInterest = this.model.get('areaOfInterest');
+
         if (areaOfInterest) {
             var layer = new L.GeoJSON(areaOfInterest);
             this._leafletMap.fitBounds(layer.getBounds(), { reset: true });
         }
-    }
+    },
+
+    updateCurrentZoomLevel: function(e) {
+        var layerMaxZoom = e.layer.options.maxZoom,
+            map = this,
+            currentZoom = map.getZoom();
+
+        // Zoom the user out to the maximum zoom of the new layer
+        // so that they don't see gray tiles.
+        if (layerMaxZoom < currentZoom) {
+            // TODO: There is no event we can consistently listen for
+            // to know when we can call this successfully.
+            // This is fixed in Leaflet 1.0 by using map.setMaxZoom
+            window.setTimeout(function() {
+                map.setZoom(layerMaxZoom);
+            }, 100);
+        }
+    },
 
+    // The max zoom for a Google layer that uses satellite imagery
+    // changes based on the location.
+    updateGoogleMaxZoom: function(e) {
+        var self = this,
+            map = e.target,
+            center = map.getCenter(),
+            latLng = { lat: center.lat, lng: center.lng }; // Google LatLng literal
+
+        this._googleMaxZoomService.getMaxZoomAtLatLng(latLng, function(response) {
+            if (response.status !== self._googleMaps.MaxZoomStatus.OK) {
+                // Leave the max layer zoom as is
+                return;
+            } else {
+                // Set layer zoom level to the max for the current area
+                _.each(self.baseLayers, function(layer) {
+                    // Needs to be updated if other Google Base maps are added
+                    if (layer._type === 'HYBRID') {
```

Comment:
As far as I can tell from the [Google docs](https://developers.google.com/maps/documentation/javascript/3.exp/reference#MaxZoomService), it's just satellite based layers. Unfortunately, we can't add any custom properties to the Leaflet layer object, but I'll add the other map type that uses imagery.